File: src/MaintenanceSystem.Domain/Entities/MaintenanceDocument.cs
namespace MaintenanceSystem.Domain.Entities;

public class MaintenanceDocument
{
    public Guid Id { get; private set; }
    public Guid WorkOrderId { get; private set; }
    public string FileName { get; private set; } = string.Empty;
    public string ContentType { get; private set; } = string.Empty;
    public string StoragePath { get; private set; } = string.Empty;
    public long FileSizeBytes { get; private set; }
    public string? Notes { get; private set; }
    public DateTime UploadedAt { get; private set; }
    public string UploadedBy { get; private set; } = string.Empty;

    private MaintenanceDocument() { }

    public static MaintenanceDocument Create(
        Guid workOrderId, string fileName, string contentType,
        string storagePath, long fileSizeBytes, string? notes, string uploadedBy) => new()
    {
        Id = Guid.NewGuid(),
        WorkOrderId = workOrderId,
        FileName = fileName,
        ContentType = contentType,
        StoragePath = storagePath,
        FileSizeBytes = fileSizeBytes,
        Notes = notes,
        UploadedAt = DateTime.UtcNow,
        UploadedBy = uploadedBy
    };
}

File: app/lib/models/maintenance_document.dart
class MaintenanceDocument {
  final String id;
  final String workOrderId;
  final String fileName;
  final String contentType;
  final int fileSizeBytes;
  final String? notes;
  final String uploadedAt;
  final String uploadedBy;

  const MaintenanceDocument({
    required this.id,
    required this.workOrderId,
    required this.fileName,
    required this.contentType,
    required this.fileSizeBytes,
    this.notes,
    required this.uploadedAt,
    required this.uploadedBy,
  });

  factory MaintenanceDocument.fromJson(Map<String, dynamic> j) => MaintenanceDocument(
        id: j['id'] as String,
        workOrderId: j['workOrderId'] as String,
        fileName: j['fileName'] as String,
        contentType: j['contentType'] as String,
        fileSizeBytes: j['fileSizeBytes'] as int,
        notes: j['notes'] as String?,
        uploadedAt: j['uploadedAt'] as String,
        uploadedBy: j['uploadedBy'] as String,
      );
}
